packages/ruleset/src/functions/required-property.js

```diff
@@ -138,24 +105,28 @@ function checkRequiredProperties(schema, path) {
  * @param {*} path the jsonpath location of "schema"
  * @returns
  */
-function checkRequired(schema, propNames, path) {
-  const errors = [];
-
-  if (Array.isArray(schema.required)) {
+function checkRequired(schema, path, subschema) {
+  // In the case of an allOf, we want to check the individual subschemas
+  // required list against the whole schema
+  const required = subschema ? subschema.required : schema.required;
+  if (Array.isArray(required)) {
+    const errors = [];
     const localPath = [...path, 'required'];
     logger.debug(`${ruleId}: checking ${localPath.join('.')}`);
 
-    schema.required.forEach(function (name) {
+    required.forEach(name => {
       logger.debug(`${ruleId}: looking for property ${name}`);
-      if (!propNames.includes(name)) {
+      if (!schemaHasProperty(schema, name)) {
```

Comment:
This is the main change - using `schemaHasProperty`, which checks recursively that at least one allOf element or all oneOf/anyOf elements contain the property